src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -0,0 +1,248 @@
+        Whitebox.setInternalState(assignmentSubmitter, "attendanceCourseService", attendanceCourseService);
+        Whitebox.setInternalState(assignmentSubmitter, "attendanceService", attendanceService);
+        Whitebox.setInternalState(assignmentSubmitter, "canvasApiWrapperService", canvasApiWrapperService);
+        Whitebox.setInternalState(assignmentSubmitter, "assignmentValidator", assignmentValidator);
+
+
+
+    }
+
+    @Test
+    public void submitCourseAttendancesHappyPath() throws IOException, AttendanceAssignmentException {
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_1_ID)).thenReturn(section1);
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_2_ID)).thenReturn(section2);
+        when(assignmentService.findBySection(section1)).thenReturn(attendanceAssignment1);
+        when(assignmentService.findBySection(section2)).thenReturn(attendanceAssignment2);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+        when(attendanceCourseService.findByCanvasCourseId(eq(COURSE_ID))).thenReturn(course);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_1_ID)).thenReturn(studentCommentsMap1);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_2_ID)).thenReturn(studentCommentsMap2);
+        when(canvasApiWrapperService.gradeMultipleSubmissionsBySection(any(), any())).thenReturn(progressOptional);
+        when(assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment1, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment1);
+        when(assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment1, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment1);
+        when(assignmentValidator.validateConfigurationSetupExistence(attendanceSummaryModel1, attendanceAssignment1)).thenReturn(attendanceAssignment1);
+        when(assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment2, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment2);
+        when(assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment2, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment2);
+        when(assignmentValidator.validateConfigurationSetupExistence(attendanceSummaryModel2, attendanceAssignment2)).thenReturn(attendanceAssignment2);
+
+        assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
+        verify(canvasApiWrapperService, times(2)).gradeMultipleSubmissionsBySection(any(), any());
+    }
+
+    @Test
+    public void submitCourseAttendancesFailCanvasGradingError() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)